Source/WebCore/animation/CSSPropertyAnimation.cpp

```diff
@@ -1433,20 +1404,59 @@ class PropertyWrapperFilter final : public PropertyWrapper<const FilterOperation
             ;
     }
 
-    bool canInterpolate(const RenderStyle& from, const RenderStyle& to, CompositeOperation) const final
+    bool canInterpolate(const RenderStyle& from, const RenderStyle& to, CompositeOperation compositeOperation) const final
     {
+        auto& fromFilterOperations = value(from);
+        auto& toFilterOperations = value(to);
+
         // https://drafts.fxtf.org/filter-effects/#interpolation-of-filters
-        auto containsReference = [](auto& filterOperations) {
+
+        auto listContainsReference = [](auto& filterOperations) {
             return filterOperations.operations().findIf([](auto& filterOperation) {
                 return filterOperation->type() == FilterOperation::OperationType::REFERENCE;
             }) != notFound;
         };
-        return !containsReference(from.filter()) && !containsReference(to.filter());
+
+        if (listContainsReference(fromFilterOperations) || listContainsReference(toFilterOperations))
+            return false;
+
+        // If one filter is none and the other is a <filter-value-list> without <url>
+        auto oneListIsEmpty = [&]() {
+            return fromFilterOperations.isEmpty() != toFilterOperations.isEmpty();
+        };
+
+        auto listsMatchUpToIndex = [&](auto maxIndex) {
```

Comment:
Might be clearer to say `numItems` instead of `maxIndex` to avoid any confusion about end-inclusiveness.

Reply:
Will change when landing.